src/core/jsdoc-tsd-parser.ts

```diff
@@ -762,6 +775,24 @@ export class JSDocTsdParser {
 		return dom.create.interface(jsdocItem.name);
 	}
 
+	private parseConstant(jsdocItem: IMemberDoclet) {
+		if (jsdocItem.isEnum) {
+			throw new Error(`item ${jsdocItem.longname} is an enum`);
+		}
+
+		let propertyType: dom.Type = dom.type.any;
+		if (jsdocItem.type && jsdocItem.type.names.length > 0) {
+			let domTypes: dom.Type[] = [];
+			jsdocItem.type.names.forEach((typeName) => {
+				domTypes.push(this.mapVariableType(typeName));
+			});
+
+			propertyType = dom.create.union(domTypes);

```

Comment:
Take a look at line 8541 (parseTypeDefinition). There is already a function to create union types